### PR TITLE
the text-package blog

### PR DESCRIPTION
**Pull request by OscarKjell**
Proposed blog post about the text-package.

### CODE REVIEW

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+title: "The text-package"
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:2]: How about 

"Introducing the text package"

for a title?

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+description: |
+  The [text](http://r-text.org/)-package attempts to provide user-friendly access and pipelines to HuggingFace's transformer language models in R.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:4]: Unfortunately, you cannot use this kind of formatting in the abstract (see preview for how it does not display correctly).
I'd just go with 
"The text package" (no hyphen)

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    affiliation_url: https://portal.research.lu.se/en/persons/oscar-kjell
+  - name: Salvatore Giorgi 
+    url: http://www.salvatoregiorgi.com/
+    affiliation: University of Pennsylvania
+    affiliation_url: http://www.salvatoregiorgi.com/
+  - name: H Andrew Schwartz  
+    url: https://www3.cs.stonybrook.edu/~has/
+    affiliation: Stony Brook University
+    affiliation_url: https://www3.cs.stonybrook.edu/~has/
+categories:
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:18]: Please only make use of existing categories :-)
From these, only 

Natural Language Processing 

exists. Plus, we have 

R

as a category always (if R-related), for a technical reason.

Thanks!

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  - Natural Language Processing
+  - Machine Learning
+  - Transformers
+  - HuggingFace
+date: 09-29-2022
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:23]: please change to Tuesday, then :-)

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+output: distill::distill_article
+preview: img/ai_blog_word_plot.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE,
+                      eval = FALSE)
+```
+
+AI-based language analysis has recently gone through a “paradigm shift” (Bommasani, et al., 2021, p. 1) thanks in part to a new technique referred to as transformer language model (Vaswani et al., 2017, Liu et al., 2019). Companies, including Google, Meta, and OpenAI have released such models including BERT, RoBERTa, and GPT, that have achieved unprecedented large improvements across most language tasks such as web search and sentiment analysis. While these language models are accessible in Python, and for typical AI tasks through [HuggingFace](https://huggingface.co/), the R-package [text](http://r-text.org/) makes HuggingFace and state-of-the-art transformer language models accessible as social scientific pipelines in R.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:35]: GH suggestions for some reason does not work here, so I'm entering things manually :-)

Bommasani, et al., -> Bommasani et al.,

thanks -> , thanks

including -> , including

R-package -> R package

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+
+## Introduction
+
+We developed the `text`-package [(Kjell, Giorgi & Schwartz, 2022)](https://psyarxiv.com/293kt) with two objectives in mind:
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:40]: `text`-package -> `text` package

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+
+## Introduction
+
+We developed the `text`-package [(Kjell, Giorgi & Schwartz, 2022)](https://psyarxiv.com/293kt) with two objectives in mind: 
+To serve as a modular solution for downloading and using transformer language models. This, for example, includes transforming text to word embeddings as well as accessing common language model tasks such as text classification, sentiment analysis, text generation, question answering, translation and so on. 
+To provide an end-to-end solution that is catered for human-level analyses including pipelines for state-of-the-art AI techniques tailored for predicting characteristics of the person that produced the language or eliciting insights about linguistic correlates of psychological attributes.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:42]: catered -> designed

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+This blog post shows how to install the `text`-package, transforming text to state-of-the-art contextual word embeddings, using language analysis tasks as well as visualizing words in word embedding space.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:44]: This blog post shows how to install the `text`-package, transforming text to state-of-the-art contextual word embeddings, using language analysis tasks as well as visualizing words in word embedding space.

This blog post shows how to install the `text` package, transform text to state-of-the-art contextual word embeddings, use language analysis tasks as well as visualize words in word embedding space.

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+To serve as a modular solution for downloading and using transformer language models. This, for example, includes transforming text to word embeddings as well as accessing common language model tasks such as text classification, sentiment analysis, text generation, question answering, translation and so on. 
+To provide an end-to-end solution that is catered for human-level analyses including pipelines for state-of-the-art AI techniques tailored for predicting characteristics of the person that produced the language or eliciting insights about linguistic correlates of psychological attributes.
+
+This blog post shows how to install the `text`-package, transforming text to state-of-the-art contextual word embeddings, using language analysis tasks as well as visualizing words in word embedding space.  
+
+## Installation and setting up a python environment
+The `text`-package is setting up a python environment to get access to the HuggingFace language models. The first time after installing the `text`-package you need to run two functions: `textrpp_install()` and `textrpp_initialize()`.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:47]: `text`-package without hyphen

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+generated_text <- textGeneration("The meaning of life is")
+generated_text
+
+```
+
+For more examples of available language model tasks, for example, see [textSum()](https://www.r-text.org/reference/textSum.html), [textQA()](https://www.r-text.org/reference/textQA.html), [textTranslate()](https://www.r-text.org/reference/textTranslate.html), and [textZeroShot()](https://www.r-text.org/reference/textZeroShot.html) under [Language Analysis Tasks](https://www.r-text.org/reference/index.html).
+
+
+# Visualizing Words in Embedding Space
+Visualizing words in the `text`-package is achieved in two steps: First with a function to pre-process the data, and second to plot the words including adjusting visual characteristics such as color and font size.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:103]: `text`-package (see also elsewhere please)

File: _posts/2022-09-29-r-text/r-text.Rmd [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+
+# Visualizing Words in Embedding Space
+Visualizing words in the `text`-package is achieved in two steps: First with a function to pre-process the data, and second to plot the words including adjusting visual characteristics such as color and font size. 
+To demonstrate these two functions we use example data included in the `text`-package: `Language_based_assessment_data_3_100`. We show how to create a two-dimensional figure with words that individuals have used to describe their harmony in life plotted according to two different well-being questionnaires: the harmony in life scale and the satisfaction with life scale. So, the x-axis shows words that are related to low versus high harmony in life scale scores, and the y-axis shows words related to low versus high satisfaction with life scale scores.
```
Comment (skeydan) [inline _posts/2022-09-29-r-text/r-text.Rmd:104]: in life plotted -> in life, plotted

**Review by skeydan — COMMENTED** [review]
Thanks, that looks great!
I've left a few suggestions.

Once you push edits, you can preview the generated html from the actions tab in GH - for example, for the initial run you'd find the downloadable artifact on the bottom of this page:

image

If you (in the preview) navigate to the blog homepage (index.html), you'll also see that formatting does not work there (and so, has to be removed from the abstract overall).

Thanks!

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 1 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:2]: sounds great, thanks

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 2 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:4]: it's been removed.

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 3 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:18]: removed, and included R

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 4 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:23]: changed :)

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 5 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:35]: done

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 6 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:40]: done

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 7 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:42]: agree, thanks

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 9 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:47]: addressed everywhere

File: _posts/2022-09-29-r-text/r-text.Rmd (hunk 11 above)
Comment (OscarKjell) [reply _posts/2022-09-29-r-text/r-text.Rmd:104]: thanks

**Comment by skeydan** [comment]
Thanks, looks great! Will merge on Tuesday!

**Pull request merged**